src/timeMachine/components/dynamicFluxFunctionsToolbar/FluxFunctionsToolbar.tsx

```diff
@@ -8,56 +9,105 @@ import {DapperScrollbars} from '@influxdata/clockface'
+  const [fluxServiceError, setFluxServiceError] = useState<RemoteDataState>(
+    RemoteDataState.NotStarted
+  )
 
   const handleSearch = (searchTerm: string): void => {
     setSearchTerm(searchTerm)
   }
 
+  useEffect(() => {
+    const getFluxFuncs = async () => {
+      try {
+        setFluxServiceError(RemoteDataState.Loading)
+
+        if (props.fluxFunctions.length === 0) {
+          await props.getFluxPackages()
+          setFluxServiceError(RemoteDataState.Done)
+        }
+        setFluxServiceError(RemoteDataState.Done)
+      } catch (err) {
+        console.error(err)
+        setFluxServiceError(RemoteDataState.Error)
+      }
+    }
+    getFluxFuncs()
+  }, [])
+
   const handleClickFunction = useCallback(
-    (func: FluxToolbarFunction) => {
+    (func: Fluxdocs) => {
       props.onInsertFluxFunction(func)
     },
     [props.onInsertFluxFunction]
   )
 
   return useMemo(() => {
     return (
-      <ErrorBoundary>
-        <FluxToolbarSearch onSearch={handleSearch} resourceName="Functions" />
-        <DapperScrollbars className="flux-toolbar--scroll-area">
-          <div className="flux-toolbar--list" data-testid="flux-toolbar--list">
-            <TransformToolbarFunctions
-              funcs={FLUX_FUNCTIONS}
-              searchTerm={searchTerm}
+      <SpinnerContainer
+        loading={fluxServiceError}
+        spinnerComponent={<TechnoSpinner />}
+      >
+        <ErrorBoundary>
+          <FluxToolbarSearch onSearch={handleSearch} resourceName="Functions" />
+          <DapperScrollbars className="flux-toolbar--scroll-area">
+            <div
+              className="flux-toolbar--list"
+              data-testid="flux-toolbar--list"
             >
-              {sortedFunctions =>
-                sortedFunctions.map(func => (
-                  <ToolbarFunction
-                    onClickFunction={handleClickFunction}
-                    key={`${func.name}_${func.desc}`}
-                    func={func}
-                    testID={func.name}
-                  />
-                ))
-              }
-            </TransformToolbarFunctions>
-          </div>
-        </DapperScrollbars>
-      </ErrorBoundary>
+              <TransformToolbarFunctions
+                funcs={props.fluxFunctions}
+                searchTerm={searchTerm}
+              >
+                {sortedFunctions =>
+                  sortedFunctions.map((func, index) => (
+                    <ToolbarFunction
+                      onClickFunction={handleClickFunction}
+                      key={index}
+                      func={func}
+                      testID={func.name}
+                    />
+                  ))
+                }
+              </TransformToolbarFunctions>
+            </div>
+          </DapperScrollbars>
+        </ErrorBoundary>
+      </SpinnerContainer>
     )
-  }, [searchTerm, handleClickFunction])
+  }, [searchTerm, handleClickFunction, fluxServiceError])
```

Comment:
Add `fluxFunctions` to this array, along with the changes mentioned above.

Also, alphabetize if you get the chance and remember, but don't worry if not.

Reply:
oh good catch! added and alphabetized